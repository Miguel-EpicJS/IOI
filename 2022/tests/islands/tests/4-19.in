4e5150f5a11b21d9ac3b44f2259e7a3452343ad4
1000 2000
72 799
72 799
168 678
168 678
635 981
635 981
857 678
857 678
183 575
183 575
209 480
209 480
83 967
83 967
712 534
712 534
367 678
367 678
723 2
723 2
100 39
100 39
431 123
431 123
857 678
857 678
974 749
974 749
349 620
349 620
327 887
327 887
222 921
222 921
0 678
0 678
919 247
919 247
893 534
893 534
653 678
653 678
824 141
824 141
128 307
128 307
469 734
469 734
337 678
337 678
737 0
737 0
689 678
689 678
53 724
53 724
618 0
618 0
258 456
258 456
471 773
471 773
417 511
417 511
495 678
495 678
198 761
198 761
168 276
168 276
619 801
619 801
484 508
484 508
161 964
161 964
612 678
612 678
943 231
943 231
263 512
263 512
168 0
168 0
57 961
57 961
118 849
118 849
152 917
152 917
936 707
936 707
233 0
233 0
521 887
521 887
580 525
580 525
920 177
920 177
53 643
53 643
137 678
137 678
558 157
558 157
676 815
676 815
624 678
624 678
103 755
103 755
937 85
937 85
737 7
737 7
306 231
306 231
920 846
920 846
38 0
38 0
851 808
851 808
257 90
257 90
668 816
668 816
771 678
771 678
49 567
49 567
292 715
292 715
649 145
649 145
90 47
90 47
913 922
913 922
535 315
535 315
426 266
426 266
612 102
612 102
574 319
574 319
151 388
151 388
137 678
137 678
217 395
217 395
367 0
367 0
658 10
658 10
71 820
71 820
899 386
899 386
415 315
415 315
384 659
384 659
728 309
728 309
469 4
469 4
712 473
712 473
917 0
917 0
482 678
482 678
135 678
135 678
438 85
438 85
560 0
560 0
891 603
891 603
794 330
794 330
975 678
975 678
616 64
616 64
759 608
759 608
979 762
979 762
331 678
331 678
219 257
219 257
516 960
516 960
482 0
482 0
39 678
39 678
855 670
855 670
600 876
600 876
235 362
235 362
484 590
484 590
137 0
137 0
648 845
648 845
179 0
179 0
482 678
482 678
758 146
758 146
991 271
991 271
183 431
183 431
653 79
653 79
880 987
880 987
291 741
291 741
477 862
477 862
737 678
737 678
400 604
400 604
704 762
704 762
633 651
633 651
675 783
675 783
74 985
74 985
915 702
915 702
155 328
155 328
389 19
389 19
70 223
70 223
940 292
940 292
291 979
291 979
771 678
771 678
516 679
516 679
347 795
347 795
192 374
192 374
451 678
451 678
75 409
75 409
949 88
949 88
563 338
563 338
624 678
624 678
572 801
572 801
219 557
219 557
767 678
767 678
919 57
919 57
430 660
430 660
576 163
576 163
862 678
862 678
911 611
911 611
787 608
787 608
317 836
317 836
680 504
680 504
500 258
500 258
262 562
262 562
450 664
450 664
799 26
799 26
653 0
653 0
609 230
609 230
880 608
880 608
624 678
624 678
737 0
737 0
29 833
29 833
234 814
234 814
852 98
852 98
846 339
846 339
380 60
380 60
665 682
665 682
343 396
343 396
79 786
79 786
43 512
43 512
915 674
915 674
361 454
361 454
851 57
851 57
14 564
14 564
618 0
618 0
758 842
758 842
240 35
240 35
417 379
417 379
484 717
484 717
364 8
364 8
295 385
295 385
234 678
234 678
135 678
135 678
367 977
367 977
737 0
737 0
183 633
183 633
290 639
290 639
564 960
564 960
895 832
895 832
740 508
740 508
290 421
290 421
26 94
26 94
103 26
103 26
806 557
806 557
593 9
593 9
654 761
654 761
137 0
137 0
868 662
868 662
176 69
176 69
826 221
826 221
183 685
183 685
581 637
581 637
324 420
324 420
500 844
500 844
253 266
253 266
525 963
525 963
862 0
862 0
151 281
151 281
792 846
792 846
641 588
641 588
947 557
947 557
264 309
264 309
298 530
298 530
102 0
102 0
873 150
873 150
890 252
890 252
203 835
203 835
419 21
419 21
831 989
831 989
906 714
906 714
83 60
83 60
771 0
771 0
152 678
152 678
26 328
26 328
827 939
827 939
392 13
392 13
186 194
186 194
553 382
553 382
80 499
80 499
484 189
484 189
733 256
733 256
875 110
875 110
873 392
873 392
422 207
422 207
896 579
896 579
159 135
159 135
158 178
158 178
606 148
606 148
212 3
212 3
55 537
55 537
367 0
367 0
466 983
466 983
159 678
159 678
208 598
208 598
959 866
959 866
58 0
58 0
39 0
39 0
161 104
161 104
423 314
423 314
409 526
409 526
136 784
136 784
596 26
596 26
361 368
361 368
538 762
538 762
992 571
992 571
561 461
561 461
696 141
696 141
414 345
414 345
618 0
618 0
491 227
491 227
988 757
988 757
191 881
191 881
244 222
244 222
479 833
479 833
599 0
599 0
814 678
814 678
236 817
236 817
930 113
930 113
208 862
208 862
442 840
442 840
79 0
79 0
618 0
618 0
355 297
355 297
104 534
104 534
872 372
872 372
899 784
899 784
618 0
618 0
690 410
690 410
680 0
680 0
514 443
514 443
7 0
7 0
709 352
709 352
334 528
334 528
219 546
219 546
296 302
296 302
809 611
809 611
968 62
968 62
913 90
913 90
874 286
874 286
482 0
482 0
357 452
357 452
152 0
152 0
929 555
929 555
950 695
950 695
550 87
550 87
334 855
334 855
176 344
176 344
321 286
321 286
823 645
823 645
867 216
867 216
940 0
940 0
862 678
862 678
71 310
71 310
715 975
715 975
540 117
540 117
168 0
168 0
977 159
977 159
869 703
869 703
386 678
386 678
73 986
73 986
960 749
960 749
191 840
191 840
7 491
7 491
23 865
23 865
186 330
186 330
988 639
988 639
142 617
142 617
633 957
633 957
11 313
11 313
219 918
219 918
337 495
337 495
642 55
642 55
690 279
690 279
414 569
414 569
460 510
460 510
779 328
779 328
670 667
670 667
83 900
83 900
215 76
215 76
392 235
392 235
479 751
479 751
482 940
482 940
181 987
181 987
377 0
377 0
438 800
438 800
320 774
320 774
979 444
979 444
92 9
92 9
131 511
131 511
219 501
219 501
696 125
696 125
377 857
377 857
191 86
191 86
265 82
265 82
578 388
578 388
897 0
897 0
389 316
389 316
386 678
386 678
814 678
814 678
794 372
794 372
433 785
433 785
70 155
70 155
874 916
874 916
683 825
683 825
851 234
851 234
930 833
930 833
657 766
657 766
770 268
770 268
573 70
573 70
164 228
164 228
975 0
975 0
292 678
292 678
20 511
20 511
520 834
520 834
522 754
522 754
516 57
516 57
382 217
382 217
734 793
734 793
767 678
767 678
885 247
885 247
22 768
22 768
945 754
945 754
524 66
524 66
456 643
456 643
624 0
624 0
766 548
766 548
315 503
315 503
324 138
324 138
576 829
576 829
991 26
991 26
786 0
786 0
227 386
227 386
435 449
435 449
680 678
680 678
300 429
300 429
975 678
975 678
161 614
161 614
593 435
593 435
955 611
955 611
103 195
103 195
678 367
678 367
589 99
589 99
671 907
671 907
173 127
173 127
386 678
386 678
769 74
769 74
715 678
715 678
313 555
313 555
58 678
58 678
644 558
644 558
622 450
622 450
816 463
816 463
968 556
968 556
959 682
959 682
367 0
367 0
758 123
758 123
624 678
624 678
411 457
411 457
792 317
792 317
749 511
749 511
86 755
86 755
249 499
249 499
912 853
912 853
688 849
688 849
366 789
366 789
744 633
744 633
67 440
67 440
647 911
647 911
277 482
277 482
994 558
994 558
786 179
786 179
814 0
814 0
686 773
686 773
167 255
167 255
394 761
394 761
491 678
491 678
500 595
500 595
862 678
862 678
79 678
79 678
752 743
752 743
131 755
131 755
618 0
618 0
187 441
187 441
108 56
108 56
482 0
482 0
241 818
241 818
587 112
587 112
183 258
183 258
874 231
874 231
782 922
782 922
58 0
58 0
984 519
984 519
794 347
794 347
355 886
355 886
786 678
786 678
955 598
955 598
906 152
906 152
468 140
468 140
560 678
560 678
865 979
865 979
796 403
796 403
135 0
135 0
384 404
384 404
136 266
136 266
751 204
751 204
802 686
802 686
421 938
421 938
96 68
96 68
754 753
754 753
492 529
492 529
646 391
646 391
396 544
396 544
800 541
800 541
875 915
875 915
847 678
847 678
839 641
839 641
857 0
857 0
41 617
41 617
55 338
55 338
377 0
377 0
595 436
595 436
135 0
135 0
418 690
418 690
786 0
786 0
6 189
6 189
117 587
117 587
244 610
244 610
758 131
758 131
239 310
239 310
389 562
389 562
603 831
603 831
1 816
1 816
485 486
485 486
952 292
952 292
179 0
179 0
15 651
15 651
376 465
376 465
83 558
83 558
613 645
613 645
415 825
415 825
423 226
423 226
712 245
712 245
166 18
166 18
625 81
625 81
136 670
136 670
897 624
897 624
168 137
168 137
925 475
925 475
653 0
653 0
568 97
568 97
650 255
650 255
769 931
769 931
520 789
520 789
707 677
707 677
227 678
227 678
370 297
370 297
653 678
653 678
168 678
168 678
438 528
438 528
131 518
131 518
494 425
494 425
516 987
516 987
942 393
942 393
268 803
268 803
15 489
15 489
918 966
918 966
353 31
353 31
295 207
295 207
711 954
711 954
591 201
591 201
44 856
44 856
994 229
994 229
179 0
179 0
377 0
377 0
552 196
552 196
478 962
478 962
417 637
417 637
367 737
367 737
891 465
891 465
928 772
928 772
539 282
539 282
191 498
191 498
308 147
308 147
712 802
712 802
248 413
248 413
973 400
973 400
745 28
745 28
934 228
934 228
165 225
165 225
897 678
897 678
135 678
135 678
345 205
345 205
988 544
988 544
618 678
618 678
418 701
418 701
159 0
159 0
15 642
15 642
730 803
730 803
491 0
491 0
442 729
442 729
709 143
709 143
756 695
756 695
648 866
648 866
630 681
630 681
482 678
482 678
277 678
277 678
875 749
875 749
219 65
219 65
79 0
79 0
158 773
158 773
336 238
336 238
199 672
199 672
857 0
857 0
94 61
94 61
153 93
153 93
604 943
604 943
767 678
767 678
737 0
737 0
164 795
164 795
705 627
705 627
161 421
161 421
267 371
267 371
320 139
320 139
176 172
176 172
126 463
126 463
959 639
959 639
155 196
155 196
343 128
343 128
767 678
767 678
806 748
806 748
662 543
662 543
823 842
823 842
287 501
287 501
540 912
540 912
771 0
771 0
788 555
788 555
605 151
605 151
888 731
888 731
209 413
209 413
490 119
490 119
896 281
896 281
30 849
30 849
38 678
38 678
814 771
814 771
847 0
847 0
322 795
322 795
812 189
812 189
38 377
38 377
164 193
164 193
896 440
896 440
862 0
862 0
364 645
364 645
797 119
797 119
173 882
173 882
324 495
324 495
181 837
181 837
460 21
460 21
695 555
695 555
58 159
58 159
411 671
411 671
294 558
294 558
851 678
851 678
917 851
917 851
605 358
605 358
560 678
560 678
650 221
650 221
106 289
106 289
771 277
771 277
576 684
576 684
799 854
799 854
21 645
21 645
995 66
995 66
364 610
364 610
909 800
909 800
483 139
483 139
447 727
447 727
184 666
184 666
291 732
291 732
337 678
337 678
979 238
979 238
649 304
649 304
656 96
656 96
496 413
496 413
52 9
52 9
332 706
332 706
308 379
308 379
291 232
291 232
665 568
665 568
519 543
519 543
736 930
736 930
607 174
607 174
58 0
58 0
337 0
337 0
495 0
495 0
645 372
645 372
169 215
169 215
418 535
418 535
612 773
612 773
39 678
39 678
705 902
705 902
744 232
744 232
314 443
314 443
367 678
367 678
729 185
729 185
53 317
53 317
369 239
369 239
842 637
842 637
430 729
430 729
43 548
43 548
862 618
862 618
624 137
624 137
609 169
609 169
767 680
767 680
206 207
206 207
927 193
927 193
771 678
771 678
406 517
406 517
337 678
337 678
159 678
159 678
683 870
683 870
495 678
495 678
960 258
960 258
495 0
495 0
984 208
984 208
403 194
403 194
975 599
975 599
794 215
794 215
770 782
770 782
668 635
668 635
866 872
866 872
612 678
612 678
208 908
208 908
277 678
277 678
56 492
56 492
495 612
495 612
407 452
407 452
124 804
124 804
888 394
888 394
92 51
92 51
752 997
752 997
857 0
857 0
135 0
135 0
683 751
683 751
39 678
39 678
221 216
221 216
407 256
407 256
728 10
728 10
420 657
420 657
39 897
39 897
953 684
953 684
187 871
187 871
929 923
929 923
95 121
95 121
756 646
756 646
479 77
479 77
414 32
414 32
135 0
135 0
398 109
398 109
550 212
550 212
782 506
782 506
897 0
897 0
11 53
11 53
896 863
896 863
786 678
786 678
710 876
710 876
364 194
364 194
539 872
539 872
715 0
715 0
765 293
765 293
415 729
415 729
186 965
186 965
573 542
573 542
618 38
618 38
491 0
491 0
845 636
845 636
135 678
135 678
857 0
857 0
874 175
874 175
409 241
409 241
582 168
582 168
653 678
653 678
952 19
952 19
179 451
179 451
233 678
233 678
300 380
300 380
82 283
82 283
918 332
918 332
493 365
493 365
547 128
547 128
994 97
994 97
696 871
696 871
88 310
88 310
824 912
824 912
792 322
792 322
582 0
582 0
597 898
597 898
891 133
891 133
492 488
492 488
72 140
72 140
689 59
689 59
839 11
839 11
233 337
233 337
605 358
605 358
829 503
829 503
789 805
789 805
277 678
277 678
262 922
262 922
23 61
23 61
482 111
482 111
280 583
280 583
896 774
896 774
311 858
311 858
11 455
11 455
572 127
572 127
403 753
403 753
478 846
478 846
71 254
71 254
724 708
724 708
640 443
640 443
505 269
505 269
340 446
340 446
919 510
919 510
179 0
179 0
650 637
650 637
672 600
672 600
487 780
487 780
398 122
398 122
480 297
480 297
633 961
633 961
159 0
159 0
275 819
275 819
878 923
878 923
75 925
75 925
425 427
425 427
58 0
58 0
186 548
186 548
7 0
7 0
993 129
993 129
915 374
915 374
624 593
624 593
347 507
347 507
447 88
447 88
948 849
948 849
618 0
618 0
725 372
725 372
917 122
917 122
179 0
179 0
238 47
238 47
854 160
854 160
335 987
335 987
477 593
477 593
230 625
230 625
479 311
479 311
599 678
599 678
161 465
161 465
875 778
875 778
36 173
36 173
290 140
290 140
908 507
908 507
862 678
862 678
211 686
211 686
479 422
479 422
505 86
505 86
792 739
792 739
445 28
445 28
765 432
765 432
241 727
241 727
680 847
680 847
285 236
285 236
943 543
943 543
451 678
451 678
607 848
607 848
862 678
862 678
680 0
680 0
7 678
7 678
922 134
922 134
6 52
6 52
49 328
49 328
331 728
331 728
560 0
560 0
568 738
568 738
794 891
794 891
304 421
304 421
771 678
771 678
573 851
573 851
794 734
794 734
448 556
448 556
227 678
227 678
779 42
779 42
168 678
168 678
209 294
209 294
759 806
759 806
222 657
222 657
221 976
221 976
324 619
324 619
39 0
39 0
809 761
809 761
405 890
405 890
610 637
610 637
135 40
135 40
271 111
271 111
292 0
292 0
100 581
100 581
244 393
244 393
624 678
624 678
977 678
977 678
671 562
671 562
75 363
75 363
663 307
663 307
977 678
977 678
350 216
350 216
26 66
26 66
633 635
633 635
857 653
857 653
430 304
430 304
495 678
495 678
991 446
991 446
792 684
792 684
11 576
11 576
539 818
539 818
247 12
247 12
128 571
128 571
273 531
273 531
618 678
618 678
721 820
721 820
618 0
618 0
277 0
277 0
599 678
599 678
826 383
826 383
652 99
652 99
881 595
881 595
370 955
370 955
520 42
520 42
466 363
466 363
451 582
451 582
898 463
898 463
782 552
782 552
265 518
265 518
733 963
733 963
599 678
599 678
167 461
167 461
154 552
154 552
907 783
907 783
320 0
320 0
415 505
415 505
495 678
495 678
109 307
109 307
718 48
718 48
599 0
599 0
165 155
165 155
553 354
553 354
364 200
364 200
36 940
36 940
800 5
800 5
563 652
563 652
158 115
158 115
552 584
552 584
239 706
239 706
15 771
15 771
723 175
723 175
586 270
586 270
989 664
989 664
824 717
824 717
208 625
208 625
908 766
908 766
782 805
782 805
577 358
577 358
181 220
181 220
923 766
923 766
264 498
264 498
584 848
584 848
22 98
22 98
102 560
102 560
263 766
263 766
403 378
403 378
137 862
137 862
191 427
191 427
824 686
824 686
451 0
451 0
178 907
178 907
760 911
760 911
605 713
605 713
779 359
779 359
827 828
827 828
39 678
39 678
959 43
959 43
60 3
60 3
192 518
192 518
560 39
560 39
844 739
844 739
135 0
135 0
159 678
159 678
771 678
771 678
102 0
102 0
83 16
83 16
159 678
159 678
930 18
930 18
216 614
216 614
560 456
560 456
14 758
14 758
897 678
897 678
947 396
947 396
897 678
897 678
482 358
482 358
880 571
880 571
36 117
36 117
599 767
599 767
466 55
466 55
604 195
604 195
135 58
135 58
940 678
940 678
376 249
376 249
572 116
572 116
933 776
933 776
891 733
891 733
736 312
736 312
682 359
682 359
614 307
614 307
770 409
770 409
634 877
634 877
695 196
695 196
181 506
181 506
770 141
770 141
135 678
135 678
154 42
154 42
11 651
11 651
647 632
647 632
2 785
2 785
371 741
371 741
186 128
186 128
715 0
715 0
120 662
120 662
441 81
441 81
24 893
24 893
39 0
39 0
950 486
950 486
771 0
771 0
152 678
152 678
85 345
85 345
102 0
102 0
756 330
756 330
389 655
389 655
118 783
118 783
386 152
386 152
232 604
232 604
24 972
24 972
535 237
535 237
847 233
847 233
646 435
646 435
353 764
353 764
851 678
851 678
909 931
909 931
459 983
459 983
