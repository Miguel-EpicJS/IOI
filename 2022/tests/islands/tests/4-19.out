91041670623a69783795ed82bda33426101ce311
OK
1
8
34 774 16 34 35 16 774 35
